common/src/main/java/org/conscrypt/HostnameVerifier.java

```diff
@@ -0,0 +1,18 @@
+package org.conscrypt;
+
+import javax.net.ssl.SSLSession;
+
+/**
+ * This interface is used to implement hostname verification in Conscrypt.  Unlike with
+ * {@link javax.net.ssl.HostnameVerifier}, the hostname verifier is called whenever hostname
+ * verification is needed, without any use of default rules.
+ */
+public interface HostnameVerifier {
```

Comment:
I imagine most implementations of this interface will delegate to a `javax.net.ssl.HostnameVerifier`. If we use a different name (perhaps `ConscryptHostnameVerifier`), implementations will be easier to understand at a glance.

Reply:
Done.